62_CAD/shaders/main_pipeline/fragment_shader.hlsl

```diff
@@ -1,655 +1,1104 @@
-                It's a bit complex when it comes to trilinear filtering (LINEAR mip sampling), but it works in practice!
-                
-                Alternatively you can think of it as doing this instead:
-                localAlpha = smoothstep(+globals.antiAliasingFactor / exp2(max(mipLevel,0.0)), 0.0, msdf);
-                Which is reducing the aa feathering as we go up the mip levels. 
-                to avoid aa feathering of the MAX_MSDF_DISTANCE_VALUE to be less than aa factor and eventually color it and cause greyed out area around the main glyph
-            */
-            msdf *= exp2(max(mipLevel,0.0));
-            
-            LineStyle style = lineStyles[mainObj.styleIdx];
-            const float screenPxRange = input.getFontGlyphPxRange() / MSDFPixelRangeHalf;
-            const float bolden = style.worldSpaceLineWidth * screenPxRange; // worldSpaceLineWidth is actually boldenInPixels, aliased TextStyle with LineStyle
-            localAlpha = smoothstep(+globals.antiAliasingFactor / 2.0f + bolden, -globals.antiAliasingFactor / 2.0f + bolden, msdf);
-        }
-    }
-    else if (objType == ObjectType::IMAGE) 
-    {
-        const float2 uv = input.getImageUV();
-        const uint32_t textureId = input.getImageTextureId();
-
-        if (textureId != InvalidTextureIdx)
-        {
-            float4 colorSample = textures[NonUniformResourceIndex(textureId)].Sample(textureSampler, float2(uv.x, uv.y));
-            textureColor = colorSample.rgb;
-            localAlpha = colorSample.a;
-        }
-    }
-
-    uint2 fragCoord = uint2(input.position.xy);
-    
-    if (localAlpha <= 0)
-        discard;
-    
-    const bool colorFromTexture = objType == ObjectType::IMAGE;
-    
-    // TODO[Przemek]: But make sure you're still calling this, correctly calculating alpha and texture color.
-    // you can add 1 main object and push via DrawResourcesFiller like we already do for other objects (this go in the mainObjects StorageBuffer) and then set the currentMainObjectIdx to 0 here
-    // having 1 main object temporarily means that all triangle meshes will be treated as a unified object in blending operations. 
-    return calculateFinalColor<nbl::hlsl::jit::device_capabilities::fragmentShaderPixelInterlock>(fragCoord, localAlpha, currentMainObjectIdx, textureColor, colorFromTexture);
-}
+#define FRAGMENT_SHADER_INPUT
+#include "common.hlsl"
+#include <nbl/builtin/hlsl/shapes/beziers.hlsl>
+#include <nbl/builtin/hlsl/shapes/line.hlsl>
+#include <nbl/builtin/hlsl/algorithm.hlsl>
+#include <nbl/builtin/hlsl/math/equations/quadratic.hlsl>
+#include <nbl/builtin/hlsl/math/geometry.hlsl>
+#include <nbl/builtin/hlsl/spirv_intrinsics/fragment_shader_pixel_interlock.hlsl>
+#include <nbl/builtin/hlsl/jit/device_capabilities.hlsl>
+#include <nbl/builtin/hlsl/text_rendering/msdf.hlsl>
+#include <nbl/builtin/hlsl/spirv_intrinsics/fragment_shader_barycentric.hlsl>

```

Comment:
do you use this anymoe? maybe comment it out for now and say we compute the barycentric ourselves due to dilation